### PR TITLE
Feature/course : add Create Course

### PR DESCRIPTION
**Pull request by krisnadwipayana07**
### Insert Courses success Request
image
### Insert Courses bad CategoryId Request
image
### Insert Courses bad TeacherId Request
image

### CODE REVIEW

File: controllers/courses/http.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	req := request.AddRequest{}
+	c.Bind(&req)
+
+	ctx := c.Request().Context()
+	data, err := cl.CourseUsecase.Create(ctx, req.ToDomain())
+
+	if err != nil {
+		return controllers.NewErrorResponse(c, http.StatusInternalServerError, err)
+	}
+	return controllers.NewSuccesResponse(c, data)
```
Comment (daffaalex22) [inline controllers/courses/http.go]: Kurang benar

File: controllers/courses/response/add.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+func FromDomain(domain course.Domain) AddResponse {
+	return AddResponse{
+		Id:          domain.Id,
+		Title:       domain.Title,
+		Thumbnail:   domain.Thumbnail,
+		Description: domain.Description,
+		CategoryId:  domain.CategoryId,
+		Category: categories.Domain{
+			Id:        domain.Category.Id,
```
Comment (daffaalex22) [inline controllers/courses/response/add.go]: Ini Ada yg salah

**Review by daffaalex22 — APPROVED** [review]

File: drivers/database/course/mysql.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	"gorm.io/gorm"
+)
+
+type MysqlCoursesRepository struct {
+	Conn *gorm.DB
+}
+
+func NewMysqlCategoryRepository(conn *gorm.DB) course.Repository {
+	return &MysqlCoursesRepository{
+		Conn: conn,
```
Comment (ilhamtri333) [inline drivers/database/course/mysql.go]: untuk penamaan samain aja kris db, sama file folder jangan mysql.go tapi jadiin repository.go aja biar selaras

**Review by ilhamtri333 — CHANGES_REQUESTED** [review]

File: drivers/database/course/mysql.go (hunk 3 above)
Comment (krisnadwipayana07) [reply drivers/database/course/mysql.go]: di koneksinya tu kak?

Comment (ilhamtri333) [reply drivers/database/course/mysql.go]: iyaa kris

**Review by ilhamtri333 — APPROVED** [review]

**Pull request merged**